kyo-data/shared/src/main/scala/kyo/internal/BaseKyoDataTest.scala

```diff
@@ -24,11 +26,11 @@ private[kyo] trait BaseKyoDataTest:
                 Result.panic(new RuntimeException("Compilation failed", cause))
     end typeCheck
 
-    transparent inline def typeCheckFailure(inline code: String)(inline error: String): Assertion =
+    transparent inline def typeCheckFailure(inline code: String)(inline expected: String)(using frame: Frame): Assertion =
         typeCheck(code) match
             case Result.Failure(errors) =>
-                if errors.contains(error) && !error.isEmpty() then assertionSuccess
-                else assertionFailure(s"Predicate not satisfied by $errors")
+                if errors.contains(expected) && !expected.isEmpty() then assertionSuccess
+                else assertionFailure(frame.render(Map("expected" -> expected, "actual" -> errors)))
```

Comment:
added frame rendering to make it easier to read failures and present the correct source position. It was reporting `Test.scala` as the source of failures.